zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -3070,6 +3115,75 @@ components:
       required:
         - sourceElementId
         - targetElementId
+    ModifyProcessInstanceRequest:
+      type: object
+      properties:
+        activateInstructions:
+          items:
+            $ref: "#/components/schemas/ModifyProcessInstanceActivateInstruction"
+        terminateInstructions:
+          items:
+            $ref: "#/components/schemas/ModifyProcessInstanceTerminateInstruction"
+        operationReference:
+          description: >
+            A reference key chosen by the user that will be part of all records resulting from this operation.
+            Must be > 0 if provided.
+          type: integer
+          format: int64
+          minimum: 1
+    ModifyProcessInstanceActivateInstruction:
+      type: object
+      description: |
+        Instructions describing which elements should be activated in which scopes and which variables should be created.
+      properties:
+        elementId:
+          description: The ID of the element that should be activated.
+          type: string
+        ancestorElementInstanceKey:
+          description: |
+            The key of the ancestor scope the element instance should be created in.
+            Set to -1 to create the new element instance within an existing element instance of the
```

Comment:
Which should be the default ? `-1` ?

Reply:
Exactly, yeah. That way, by default, it is created as described. We can adjust the description a little bit so that part is clear. Now I am wondering: what happens in the gRPC API if I don't provide this value? It will be a `0` by default, right?

Reply:
I think the default can also be `0`, looking at [this line of code](https://github.com/camunda/camunda/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationModifyProcessor.java#L419), right?

Reply:
Looking at the code, the `ancestorKey` is valid only if > 0. So a default value of `0` I'm not sure if make sense, it should have the same behaviour of setting the value at `-1`

Reply:
They do have the same behavior, yes. For other parameters, where this is also the case (`0` and `-1` having the same behavior in the backend), we went for `0`, since this is what the gRPC API would also do. That's why I proposed `0` as the default value for consistency 🙂

Reply:
In the [java client](https://github.com/camunda/camunda/blob/6234a31c5035b8df59300e9eec6aa7f3729aed9f/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ModifyProcessInstanceCommandImpl.java#L69), the default value of the `ancestorKey` is `-1`. I think we should use the same also for the rest api, if we want to introduce the default value

Reply:
I'm shocked! 😱😄
Interesting catch 👍 
It is a `key` field, so providing `-1` seems fair compared to `0`.
OK, let's roll with `0`, thanks for catching that with the client 👍